### PR TITLE
[Export Refactor]Adding  GemmToQLinearMatMul transform

### PR DESCRIPTION
**Pull request by corey-nm**
Original pass: https://github.com/neuralmagic/sparseml/blob/9778cec8684d3fcb683504eef88655d9a4607fd6/src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py#L447

Testing plan: unit tests

### CODE REVIEW

File: src/sparseml/exporters/transforms/utils/matching.py [hunk 1]
```diff
@@ -333,6 +333,15 @@ def _match_children(
         return False
 
     children = graph.get_node_children(node)
+    if children == [] and all(
+        all(op.startswith("Optional-") for op in ops) for ops in children_ops
```
Comment (bfineran) [inline src/sparseml/exporters/transforms/utils/matching.py:337]: we should make `"Optional-"` a constant for consistency

File: src/sparseml/exporters/transforms/gemm_to_qlinearmatmul.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+                        |
+    input               QuantizeLinear
+    |                   |
+    DequantizeLinear    DequantizeLinear
+    |                   |
+            Gemm ( with no attributes)
+            |
+            QuantizeLinear
+            |
+            NOT (DequantizeLinear -> Gemm)
```
Comment (bfineran) [inline src/sparseml/exporters/transforms/gemm_to_qlinearmatmul.py:54]: what does NOT mean here?

Comment (corey-nm) [reply src/sparseml/exporters/transforms/gemm_to_qlinearmatmul.py:54]: Yeah can probably add more documentation on that. Basically if there is a Q->DQ->Gemm after the first gemm, this transform does not apply. Unsure how to capture that.

i.e. you can have:
1. Gemm -> Q -> W/e
2.  Gemm -> Q -> DQ -> Not GEMM

File: src/sparseml/exporters/transforms/utils/matching.py (hunk 1 above)
Comment (corey-nm) [reply src/sparseml/exporters/transforms/utils/matching.py:337]: Good call, will do

Comment (bfineran) [reply src/sparseml/exporters/transforms/utils/matching.py:337]: bump

File: src/sparseml/exporters/transforms/gemm_to_qlinearmatmul.py (hunk 2 above)
Comment (bfineran) [reply src/sparseml/exporters/transforms/gemm_to_qlinearmatmul.py:54]: makes sense

**Review by bfineran — COMMENTED** [review]
lgtm pending comment

**Pull request merged**